### ISSUE TITLE
Support audio streaming directives

### ISSUE THREAD

**Issue by axis-tip** [cameronhunter/alexa-response#3]
See: https://developer.amazon.com/public/community/post/Tx1DSINBM8LUNHY/New-Alexa-Skills-Kit-ASK-Feature-Audio-Streaming-in-Alexa-Skills

Would be great to be able to do something like:

```js
    return say(speechOutput)
             .card({ title: 'My Skill', content: speechOutput })
             .directives([
                {
                   "type": "AudioPlayer.Play",
                   "playBehavior": "ENQUEUE",
                   "audioItem": {
                       "stream": {
                           "token": "this-is-the-audio-token",
                           "url": "https://my-audio-hosting-site.com/audio/sample-song.mp3",
                           "offsetInMilliseconds": 0
                       }
                   }
                }
             ]);
```

I'm happy to take a stab at this but was wondering if you had any guidance before I got started. Specifically:

1. I guess I can look at your implementation of say, card, etc here: https://github.com/cameronhunter/alexa-response/blob/master/src/index.js and implement directives the same way
2. Any thoughts on the usage syntax suggested above? i.e. should the caller craft the directive and pass it in explicitly or should there be a helper or simpler way of passing in a directive? I was thinking of starting with the raw method suggested above and making it easier later.
3. Any thoughts on how to handle the fact that there could be multiple directives (it is an array) vs just one?
4. There are also a bunch of new AudioPlayer requests. I guess I could add them here: https://github.com/cameronhunter/alexa-annotations/blob/master/src/ similar to your launch request etc. I could see factoring out all the audio player requests into a separate file and scoping the exports some way... any ideas on that?

Thanks!

**Comment by tjaffri** [comment]
Sorry, created this issue with the wrong account. I'm the actual human being interested in this. Thanks.

**Comment by cameronhunter** [comment]
Hi Taqi,

We should definitely add support for these – please feel free to give it a stab. The directives seem to have a fairly complex structure so it might be a good idea to create a builder specifically for them and add a new `addDirective` builder to the response?

My aim with the response builder is to keep things simple for the developer – formatting JSON blobs manually is error prone (and boring).

**Comment by tjaffri** [comment]
Great. Taking a look at this for the following very simple case:

```js
return say(speechOutput)
             .card({ title: 'My Skill', content: speechOutput })
             .audio({type: DirectiveType.Play, url: "https://my-audio-hosting-site.com/audio/sample-song.mp3"});
```

Two things:
1. I don't see any examples of directives other than these new Audio directives but as you noted the structure seems complex enough that if they have other directives in future it might be better to have a different builder for each family of directives.
2. It appears you can simultaneously send multiple directives... (it is an array). For now I will stick with one directive at a time since that's all their documentation talks about for now.

Any advice on how to do this would be much appreciated... if you have general thoughts, 5 minutes of your time to spew them might save me a few hours of muddling through it.

Much appreciated!

**Comment by tjaffri** [comment]
Feedback appreciated. Thanks.

**Comment by cameronhunter** [comment]
I liked your approach in the PR but I thought it would be more scalable to add a `directives` builder to `Response` instead. This allows for developers to add audio directives as well as any future directive that Amazon comes up with.

To make it easier for the developer, I've added `AudioPlayer` to the module:

```javascript
import Response, { AudioPlayer } from 'alexa-response';

Response.say('Doing stuff with audio')
        .directives(
            AudioPlayer.play({ url: 'https://foo.com/my-stream-1.mp3' }),
            AudioPlayer.enqueue({ url: 'https://foo.com/my-stream-2.mp3' })
        )
        .build();
```

or

```javascript
import Response, { AudioPlayer } from 'alexa-response';

Response.build({
    say: 'Doing stuff with audio',
    directives: [
        AudioPlayer.play({ url: 'https://foo.com/my-stream-1.mp3' }),
        AudioPlayer.enqueue({ url: 'https://foo.com/my-stream-2.mp3' })
    ]
});
```

I've merged this functionality into `master` and published it as`v3.2.0`. Thanks for your work on this!

**Comment by tjaffri** [comment]
Great, thanks!
